include/veriblock/blockchain/pop/pop_utils.hpp

```diff
@@ -0,0 +1,108 @@
+#ifndef ALT_INTEGRATION_INCLUDE_VERIBLOCK_BLOCKCHAIN_POP_POP_UTILS_HPP_
+#define ALT_INTEGRATION_INCLUDE_VERIBLOCK_BLOCKCHAIN_POP_POP_UTILS_HPP_
+
+#include <algorithm>
+
+#include "veriblock/blockchain/chain.hpp"
+#include "veriblock/validation_state.hpp"
+
+namespace altintegration {
+
+template <typename ProtectingBlockTree,
+          typename ProtectedIndex,
+          typename ProtectedChainParams>
+bool endorsementValidation(ProtectedIndex& index,
```

Comment:
function is named `endorsementValidation` but it also adds endorsement to BlockIndex... Maybe call it `checkAndAddEndorsement`?